lib/Core/SQLiteBuildDB.cpp

```diff
@@ -600,6 +571,70 @@ class SQLiteBuildDB : public BuildDB {
   }
 
   /// @}
+
+  
+private:
+  /// Inserts a key if not present and always returns keyID
+  /// Sometimes key will be inserted for a lookup operation
+  /// but that is okay because it'll be added at somepoint anyway
+  KeyID getKeyID(const KeyType& key, std::string *error_out) {
```

Comment:
We should probably introduce a new private `DBKeyID` type to prevent this accidentally being mixed

Reply:
👍 Indeed, during development of this I had a bug where that was the case.